system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -69,6 +70,49 @@ class TopSite extends React.Component {
   }
 }
 
+/**
+ * A proxy class that uses double requestAnimationFrame from
+ * componentDidMount to send an async message to parent with performance
+ * data after the paint/
+ *
+ * XXX If TopSites can be changed to extend React.PureComponent, both that
+ * and this should be too.
+ * XXX Should probably be split out into a higher-order component to wrap
+ * whatever.
+ *
+ * @class TopSitesPerfTimer
+ * @extends {React.Component}
+ */
+class TopSitesPerfTimer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.perfSvc = this.props.perfSvc || perfSvc;
+    this.sendAsyncMessage = this.props.sendAsyncMessage || window.sendAsyncMessage;
+  }
+  componentDidMount() {
+    this._onNextFrame(this._sendPaintedEvent.bind(this));
+  }
+  _onNextFrame(callback) {
+    requestAnimationFrame(() => {
+      requestAnimationFrame(callback);
```

Comment:
Maybe some explanation about why the doubling of request. Also, I'm not sure if this is actually the right time as `props.TopSites` might not actually have any data, so it might have attempted to render, but it ends up rendering nothing.